W3ChampionsStatisticService/Matches/MatchRepository.cs

```diff
@@ -162,60 +165,63 @@ private PlayerScore CreateDetail(PlayerBlizzard playerBlizzard)
+        public OnGoingMatchup? LoadOnGoingMatchForPlayer(string playerId)
         {
-            var mongoCollection = CreateCollection<OnGoingMatchup>();
-
-            return await mongoCollection
-                .Find(m => m.Team1Players.Contains(playerId) 
-                        || m.Team2Players.Contains(playerId)
-                        || m.Team3Players.Contains(playerId)
-                        || m.Team4Players.Contains(playerId)
-                )
-                .FirstOrDefaultAsync();
+            return _onGoingMatchups
+                .Find(m => m.Team1Players.Contains(playerId)
+                           || m.Team2Players.Contains(playerId)
+                           || m.Team3Players.Contains(playerId)
+                           || m.Team4Players.Contains(playerId)
+                );
         }
 
-        public Task DeleteOnGoingMatch(string matchId)
+        public bool DeleteOnGoingMatch(string matchId)
         {
-            return Delete<OnGoingMatchup>(x => x.MatchId == matchId);
+            return _onGoingMatchups.Remove(_onGoingMatchups.Find(x => x.MatchId == matchId));
         }
 
-        public async Task<List<OnGoingMatchup>> LoadOnGoingMatches(
+        public List<OnGoingMatchup> LoadOnGoingMatches(
             GameMode gameMode = GameMode.Undefined,
             GateWay gateWay = GateWay.Undefined,
             int offset = 0,
             int pageSize = 100)
         {
-            var mongoCollection = CreateCollection<OnGoingMatchup>();
-
-            var events = await mongoCollection
-                    .Find(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
-                    && (gateWay == GateWay.Undefined || m.GateWay == gateWay))
-                .SortByDescending(s => s.Id)
+            var events = _onGoingMatchups
+                .FindAll(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
+                              && (gateWay == GateWay.Undefined || m.GateWay == gateWay))
+                .OrderByDescending(s => s.Id)
                 .Skip(offset)
-                .Limit(pageSize)
-                .ToListAsync();
+                .Take(pageSize)
+                .ToList();
 
             return events;
         }
 
-        public Task<long> CountOnGoingMatches(
+        public long CountOnGoingMatches(
             GameMode gameMode = GameMode.Undefined,
             GateWay gateWay = GateWay.Undefined)
         {
-            return CreateCollection<OnGoingMatchup>()
-                .CountDocumentsAsync(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
-                && (gateWay == GateWay.Undefined || m.GateWay == gateWay));
+            return _onGoingMatchups.FindAll(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
+                                                 && (gateWay == GateWay.Undefined || m.GateWay == gateWay)).Count;
         }
     }
+
+    internal static class OnGoingMatchUps
```

Comment:
maybe rename this to OnGoingMatchesMemoryStore?

Reply:
yep, sure